t/ci/800-stage-tests.sh

```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+
+# this is a unit speed test.  It creates a randomized file
+# on the parallel file system.  Then it uses the transfer
+# API to transfer that file into the UnifyFS space, then
+# then transfers it back out again to a *different* location.
+# The validation test is to compare the final file to the
+# original file to make sure they're bit by bit identical.
+
+# env variable STAGE_FILE_SIZE_IN_MB sets the size
+# this has a default below of 100 MB
+
+# As long as UNIFY_LOG_DIR is set this script will have
+# places to put its temp files.  This includes successive
+# of this script internally keeping track of timing
+# iteration indices.
+
+# These two variables are written to every timing data line:
+# the variable STAGE_TEST_OVERALL_CONFIG is used to tag a whole set
+# of tests.
+# the variable STAGE_TEST_SPECIFIC_CONFIG tags lower-level config
+# info, typically what's changing in the overall set of tests
+# file size is taken care of, so the overall config variable
+# will typically be what machine or software is being tested.
+# The specific config might be a specific server configuration
+# that's being checked or iterated over.
+
+test_description="UnifyFS Stage-in+Stage-out tests"
+
+# utility checking to make sure everything's in place.
+STAGE_EXE=${UNIFYFS_EXAMPLES}/unifyfs-stage
+test_expect_success "unify-stage executable exists" '
+     test_path_is_file $STAGE_EXE
+'
+test_expect_success "\$UNIFYFS_LOG_DIR exists" '
+    test_path_is_dir $UNIFYFS_LOG_DIR
+'
+
+# make and check utility directories
+STAGE_CFG_DIR="${UNIFYFS_LOG_DIR}/stage/config_800"
+STAGE_LOG_DIR="${UNIFYFS_LOG_DIR}/stage/log_800"
+STAGE_SRC_DIR="${UNIFYFS_LOG_DIR}/stage/stage_source_800"
+STAGE_DST_DIR="${UNIFYFS_LOG_DIR}/stage/stage_destination_800"
+mkdir -p $STAGE_CFG_DIR $STAGE_LOG_DIR $STAGE_SRC_DIR $STAGE_DST_DIR
+
+test_expect_success "stage testing dirs exist" '
+    test_path_is_dir  $STAGE_CFG_DIR &&
+    test_path_is_dir  $STAGE_LOG_DIR &&
+    test_path_is_dir  $STAGE_SRC_DIR &&
+    test_path_is_dir  $STAGE_DST_DIR
+'
+
+# keeping track of where we are in iterations to *guarantee* even
+# with weird iteration counts that the iter counter will always
+# be unique and thus preserve data.
+STAGE_TEST_INDEX_FILE="${STAGE_LOG_DIR}/stage_test_index.txt"
+# ensure existence of index file
+# index should consist of a number
+if ! [ -e "$STAGE_TEST_INDEX_FILE" ] ; then
+    echo "1" > ${STAGE_TEST_INDEX_FILE}
+fi
+# the test index file now exists
+STAGE_TEST_INDEX=`head -n1 ${STAGE_TEST_INDEX_FILE}`
+NEW_STAGE_TEST_INDEX=`printf %d $((STAGE_TEST_INDEX+1))`
```

Comment:
Seems like this could be shortened to:
`NEW_STAGE_TEST_INDEX=$(( STAGE_TEST_INDEX + 1 ))`